Created SVG to JSON Python Script

vnguyen727 opened the pull request:
Created a python script that

1. parses svg files
2. calculate absolute coordinates
3. declares node types (halls, doors, elevator)
4. generates JSON objects to match the template format
5. organizes the nodes by type

I tested this with the first floor of the IST with the script.
python3 svg_to_graph.py istF1.svg 1

vnguyen727 committed:
Create test.py

vnguyen727 committed:
SVG to JSON Python Script

Created a Python Script that
- parses svg files
- calculates coordinates
- declares node types
- generates JSON objects
- organizes nodes and types

and tested the script with the IST first floor svg by using:

python3 svg_to_graph.py istF1.svg 1

ekuhhirte reviewed (dismissed):
looks nice

DJFreer92 reviewed (changes requested):
I do not believe there should not be any negative coordinates. The values are also all very small while I'd expect them to be more spread out. The range for the coordinates in Inkscape is 0 - ~695 for x, and 0 - ~458 for y. This may have been caused by a miscalculation of the points relative to their parent(s). You could try plotting the coordinates with (0, 0) in the top left corner as a way to verify the correct translation.

I'd suggest putting the svg_to_graph.py file in a new scripts/ directory in the root directory of the project instead of with the data.


I like your code, it has good documentation and type annotations 👍🏻

DJFreer92 committed:
Merge branch 'main' into svg-to-json

DJFreer92 committed:
Merge branch 'main' into svg-to-json

DJFreer92 committed:
new svg to graph script (needs some work still)

Signed-off-by: Daniel Freer <deletes_minimal.0w@icloud.com>

DJFreer92 committed:
updated svg to graph script

Signed-off-by: Daniel Freer <deletes_minimal.0w@icloud.com>

DJFreer92 committed:
documented node naming convention

Signed-off-by: Daniel Freer <deletes_minimal.0w@icloud.com>

DJFreer92 committed:
Merge branch 'main' into svg-to-json

DJFreer92 committed:
Merge branch 'main' into svg-to-json

Totoro51212 committed:
Merge branch 'main' into svg-to-json

Totoro51212 committed:
Implemented connections and distances

Took the output of the script and adapted it to use the connections json file in order to calculate distances between connections for navigation

- loads graph file which is outputted from node_connections_helper.py
- connects nodes together in order to calculate their distances
- saves the output as a json file for input to the navigation algorithm
- minor bug fixes

Totoro51212 commented:
Took the fixed output of the script by @DJFreer92 and adapted it to use the connections JSON file in order to calculate distances between connections for navigation

- loads graph file which is outputted from node_connections_helper.py
- connects nodes together in order to calculate their distances
- saves the output as a JSON file for input to the navigation algorithm
- minor bug fixes

Totoro51212 committed:
Delete nodes_output.json

Totoro51212 committed:
Create new nodes_output.json

Totoro51212 commented:
Removed old output file and generated a new one

reference `public\data\indoors\example\example.json` for the full output of the script
